packages/contracts-bedrock/scripts/fpac/FPACOPS.s.sol

```diff
@@ -0,0 +1,141 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+import { Proxy } from "src/universal/Proxy.sol";
+import { IDisputeGame } from "src/dispute/interfaces/IDisputeGame.sol";
+import { StdAssertions } from "forge-std/StdAssertions.sol";
+import "scripts/Deploy.s.sol";
+
+/// @notice Deploys the Fault Proof Alpha Chad contracts.
+contract FPACOPS is Deploy, StdAssertions {
+    ////////////////////////////////////////////////////////////////
+    //                        ENTRYPOINTS                         //
+    ////////////////////////////////////////////////////////////////
+
+    function deployFPAC(address _proxyAdmin, address _systemOwnerSafe) public {
+        console.log("Deploying a fresh FPAC system and OptimismPortal2 implementation.");
+
+        prankDeployment("ProxyAdmin", msg.sender);
+        prankDeployment("SystemOwnerSafe", msg.sender);
+
+        // Deploy the DisputeGameFactoryProxy.
+        deployERC1967Proxy("DisputeGameFactoryProxy");
+
+        // Deploy implementations.
+        deployDisputeGameFactory();
+        deployPreimageOracle();
+        deployMips();
+
+        // Deploy the new `OptimismPortal` implementation.
+        deployOptimismPortal2();
+
+        // Initialize the DisputeGameFactoryProxy.
+        initializeDisputeGameFactoryProxy();
+
+        // Deploy the Cannon Fault game implementation and set it as game ID = 0.
+        setCannonFaultGameImplementation({ _allowUpgrade: false });
+        // Deploy the Permissioned Cannon Fault game implementation and set it as game ID = 1.
+        setPermissionedCannonFaultGameImplementation({ _allowUpgrade: false });
+
+        // Transfer ownership of the DisputeGameFactory to the SystemOwnerSafe, and transfer the administrative rights
+        // of the DisputeGameFactoryProxy to the ProxyAdmin.
+        transferDGFOwnershipFinal({ _proxyAdmin: _proxyAdmin, _systemOwnerSafe: _systemOwnerSafe });
+
+        // Run post-deployment assertions.
+        postDeployAssertions({ _proxyAdmin: _proxyAdmin, _systemOwnerSafe: _systemOwnerSafe });
+
+        // Print overview
+        printConfigReview();
+    }
+
+    ////////////////////////////////////////////////////////////////
+    //                          HELPERS                           //
+    ////////////////////////////////////////////////////////////////
+
+    /// @notice Initializes the DisputeGameFactoryProxy with the DisputeGameFactory.
+    function initializeDisputeGameFactoryProxy() internal broadcast {
+        console.log("Initializing DisputeGameFactoryProxy with DisputeGameFactory.");
+
+        address dgfProxy = mustGetAddress("DisputeGameFactoryProxy");
+        Proxy(payable(dgfProxy)).upgradeToAndCall(
+            mustGetAddress("DisputeGameFactory"), abi.encodeWithSignature("initialize(address)", msg.sender)
```

Comment:
Something that i'd like to do longer term is to break out the calldata creation into internal funcs so they can be used by both EOAs and the safes